CMake/resolve_dependency_modules/gflags.cmake

```diff
@@ -43,7 +43,20 @@ set(GFLAGS_IS_SUBPROJECT ON)
 
 # Workaround for https://github.com/gflags/gflags/issues/277
 unset(BUILD_SHARED_LIBS)
+if(DEFINED CACHE{BUILD_SHARED_LIBS})
+  set(CACHED_BUILD_SHARED_LIBS ${BUILD_SHARED_LIBS})
+  unset(BUILD_SHARED_LIBS CACHE)
+endif()
+
 FetchContent_MakeAvailable(gflags)
+
+# Workaround for https://github.com/gflags/gflags/issues/277
+if(DEFINED CACHED_BUILD_SHARED_LIBS AND NOT DEFINED CACHE{BUILD_SHARED_LIBS})
```

Comment:
Don't we want to always reset to the value in `CACHED_BUILD_SHARED_LIBS`, as that being defined means we changed the original value/unset it?

Reply:
Would you mean to remove `AND NOT DEFINED CACHE{BUILD_SHARED_LIBS}`? It's reasonable to me if so.

I was trying to consider the case that `FetchContent_MakeAvailable(gflags)` internally caches `BUILD_SHARED_LIBS` again. But apparently if that happens, we cannot change the cached value without `FORCE` so it's anyway a no-op even jumping into line 55.

Reply:
Thanks, yes that's right. We should force it then. Otherwise it could leak into the main build, unlikely but can cause fun to debug issues ^^